### PR TITLE
Remove built event and replace usage in flying carpet.

### PR DESCRIPTION
**Pull request by cramforce**
Addresses `AmpEvents.BUILT` part of #23557

CC @Enriqe

### CODE REVIEW

**Comment by gmajoulet** [comment]
cc @Enriqe

File: extensions/amp-fx-flying-carpet/0.1/amp-fx-flying-carpet.js [hunk 1]
```diff
@@ -167,20 +166,50 @@ export class AmpFlyingCarpet extends AMP.BaseElement {
...
+  observeNewChildren_() {
+    const observer = new MutationObserver(changes => {
+      for (let i = 0; i < changes.length; i++) {
+        const {addedNodes} = changes[i];
+        if (!addedNodes) {
+          continue;
+        }
+        for (let n = 0; n < addedNodes.length; n++) {
+          const node = addedNodes[n];
+          if (!node.signals) {
```
Comment (jridgewell) [inline extensions/amp-fx-flying-carpet/0.1/amp-fx-flying-carpet.js]: Can we guarantee that the element is upgraded to a CE by now? I think we need to use `isAmpElement` to check instead, then wait for upgrade, then wait for built.

File: extensions/amp-fx-flying-carpet/0.1/amp-fx-flying-carpet.js [hunk 2]
```diff
@@ -167,20 +166,50 @@ export class AmpFlyingCarpet extends AMP.BaseElement {
...
+            return;
```
Comment (jridgewell) [inline extensions/amp-fx-flying-carpet/0.1/amp-fx-flying-carpet.js]:
```suggestion
            continue;
```

File: src/base-element.js [hunk 3]
```diff
@@ -678,8 +678,9 @@ export class BaseElement {
     attributes = isArray(attributes) ? attributes : [attributes];
     for (let i = 0; i < attributes.length; i++) {
       const attr = attributes[i];
-      if (this.element.hasAttribute(attr)) {
-        element.setAttribute(attr, this.element.getAttribute(attr));
+      const val = this.element.getAttribute(attr);
```
Comment (jridgewell) [inline src/base-element.js]: I'm assuming the rest of the changes are only meant be in https://github.com/ampproject/amphtml/pull/23767.

Comment (cramforce) [reply src/base-element.js]: Yeah.

**Comment by cramforce** [comment]
Re: Testing: Do we have a manual test case?

**Comment by jridgewell** [comment]
There's a manual one, but we can't guarantee the upgrade order to test manually. It has to be simulated via an `appendChild` in a unit test.

**Comment by cramforce** [comment]
Makes sense. I mostly wanted to ensure it works at all.

File: extensions/amp-fx-flying-carpet/0.1/amp-fx-flying-carpet.js (hunk 2 above)
Comment (cramforce) [reply extensions/amp-fx-flying-carpet/0.1/amp-fx-flying-carpet.js]: Done

File: extensions/amp-fx-flying-carpet/0.1/amp-fx-flying-carpet.js (hunk 1 above)
Comment (cramforce) [reply extensions/amp-fx-flying-carpet/0.1/amp-fx-flying-carpet.js]: I think that we can actually assume they have been upgraded yet.

Either you get the sync call from native CEs or your get an earlier MutationObserver that has done it.

**Pull request merged**